src/sentry/rules/history/preview.py

```diff
@@ -74,3 +77,42 @@ def preview(
             group_last_fire[event.group_id] = event.timestamp
 
     return Group.objects.filter(id__in=group_ids)
+
+
+def get_events(project: Project, condition_activity: Sequence[ConditionActivity]) -> Dict[str, Any]:
+    """
+    Returns events that have caused issue state changes.
+    """
+    group_ids = []
+    event_ids = []
+    for act in condition_activity:
+        if act.type == ConditionActivityType.CREATE_ISSUE:
+            group_ids.append(act.group_id)
+        elif act.type in (ConditionActivityType.REGRESSION, ConditionActivityType.REAPPEARED):
+            event_id = act.data.get("event_id", None)
+            if event_id is not None:
+                event_ids.append(event_id)
+
+    # TODO: Add more columns as more event filters are supported
+    columns = ["event_id"]
+    events_from_group_ids: Dict[str, Any] = {"data": []}
+    events_from_event_ids: Dict[str, Any] = {"data": []}
```

Comment:
can combine these into one dict and just append/update as necessary. like maybe just have an array and then append based on the `raw_query().get('data', [])`